src/sparsematrix.jl

```diff
@@ -931,6 +931,24 @@ function Matrix(S::AbstractSparseMatrixCSC{Tv}) where Tv
 end
 Array(S::AbstractSparseMatrixCSC) = Matrix(S)
 
+# Faster version for SparseMatrixCSC
+function Matrix(S::SparseMatrixCSC{Tv}) where Tv
```

Comment:
What if you overload

```julia
Base.copyto!(A::Matrix, S::SparseMatrixCSC)
```

Then you get a fast constructor and a fast `copyto!` method at the same time.

Reply:
That's a nice idea - I do have to handle the cases that A is larger than S in that case, but seems to work. 
If you think the current update is good, then I think it makes sense to use the same structure for the vector case (for which `copyto!` currently is slow as well)